wpilibj/src/shared/java/edu/wpi/first/wpilibj/CircularBuffer.java

```diff
@@ -7,11 +7,13 @@
 
 package edu.wpi.first.wpilibj;
 
+import java.util.Vector;
+
 /**
  * This is a simple circular buffer so we don't need to "bucket brigade" copy old values.
  */
 public class CircularBuffer {
-  private double[] m_data;
+  private Vector<Double> m_data;
```

Comment:
I think changing from double[] to Vector<Double> is a significant pessimization.  Won't this cause every element update to hit the heap since it's not using a native?